dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java

```diff
@@ -110,6 +109,38 @@ public QuayImageRegistry(final HttpClient client, final ObjectMapper objectMappe
         return tags;
     }
 
+    public Tag getImageInformationForTag(Tool tool, Tag tag) {
+        LOG.info(quayToken.getUsername() + " ======================= Getting image for tag {}================================", tag.getName());
+
+        final String repo = tool.getNamespace() + '/' + tool.getName();
+        Map<String, Map<String, Map<String, String>>> map = new HashMap<>(getToolFromQuay(tool));
+        if (map != null) {
+            final Map<String, String> tagInfo = map.get("tags").get(tag.getName());
+            final String manifestDigest = tagInfo.get("manifest_digest");
+            final String imageID = tagInfo.get("image_id");
+
+            List<Checksum> checksums = new ArrayList<>();
+            checksums.add(new Checksum(manifestDigest.split(":")[0], manifestDigest.split(":")[1]));
+            tag.setImages(Collections.singleton(new Image(checksums, repo, tag.getName(), imageID)));
+        }
+        return tag;
+    }
+
+    private  Map<String, Map<String, Map<String, String>>> getToolFromQuay(Tool tool) {
```

Comment:
I wouldn't necessarily change this for the PR, but I would suggest getting into the habit of returning an Optional whenever a function can return null. That way it is made clear in the signature that the caller has to check to see if the value actually exists or not.